x/tx/signing/aminojson/aminojson.go

```diff
@@ -16,14 +15,14 @@ import (
 
 // SignModeHandler implements the SIGN_MODE_LEGACY_AMINO_JSON signing mode.
 type SignModeHandler struct {
-	fileResolver protodesc.Resolver
+	fileResolver signing.ProtoFileResolver
 	typeResolver protoregistry.MessageTypeResolver
 	encoder      Encoder
 }
 
 // SignModeHandlerOptions are the options for the SignModeHandler.
 type SignModeHandlerOptions struct {
-	FileResolver protodesc.Resolver
+	FileResolver signing.ProtoFileResolver
```

Comment:
Can't we just use the file resolver on the `Encoder`? Or is this so that `Encoder` can be left `nil`?

Reply:
Right this supports he case where `Encoder` is nil